### PR TITLE
Replace string.data() with string.c_str()

### PR DESCRIPTION
**Pull request by karabowi**
Changed the macros in the propagator example,
as well as varius calls to .data() throughout the code.

Addresses issue #1296.

---

Checklist:

* [x] Rebased against `dev` branch
* [x] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [x] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: examples/MQ/histogramServer/FairMQExHistoDevice.cxx [hunk 1]
```diff
@@ -56,7 +56,7 @@ bool FairMQExHistoDevice::ConditionalRun()
     RootSerializer().Serialize(*message, &fArrayHisto);
 
     for (auto& channel : fChannels) {
-        Send(message, channel.first.data());
+        Send(message, channel.first.c_str());
```
Comment (dennisklein) [inline examples/MQ/histogramServer/FairMQExHistoDevice.cxx:59]:
```suggestion
        Send(message, channel.first);
```
Does this not work?

File: examples/MQ/pixelDetector/src/devices/FairMQRunDevice.cxx [hunk 2]
```diff
@@ -54,7 +54,7 @@ void FairMQRunDevice::SendObject(TObject* obj, const std::string& chan)
     if (Send(mess, chan) > 0) {
         if (Receive(rep, chan) > 0) {
             std::string repString{static_cast<char*>(rep->GetData()), rep->GetSize()};
-            LOG(info) << " -> " << repString.data();
+            LOG(info) << " -> " << repString.c_str();
```
Comment (dennisklein) [inline examples/MQ/pixelDetector/src/devices/FairMQRunDevice.cxx:57]:
```suggestion
            LOG(info) << " -> " << repString;
```

File: examples/MQ/pixelDetector/src/devices/FairMQRunDevice.cxx [hunk 3]
```diff
@@ -68,7 +68,7 @@ void FairMQRunDevice::SendBranches()
     TObjString* ObjStr;
 
     for (auto& mi : fChannels) {
-        LOG(debug) << "trying channel >" << mi.first.data() << "<";
+        LOG(debug) << "trying channel >" << mi.first.c_str() << "<";
```
Comment (dennisklein) [inline examples/MQ/pixelDetector/src/devices/FairMQRunDevice.cxx:71]:
```suggestion
        LOG(debug) << "trying channel >" << mi.first << "<";
```

File: examples/MQ/pixelSimSplit/src/devices/FairMQTransportDevice.cxx [hunk 4]
```diff
@@ -154,7 +154,7 @@ void FairMQTransportDevice::InitializeRun()
     if (Send(req, fUpdateChannelName) > 0) {
         if (Receive(rep, fUpdateChannelName) > 0) {
             std::string repString{static_cast<char*>(rep->GetData()), rep->GetSize()};
-            LOG(info) << " -> " << repString.data();
+            LOG(info) << " -> " << repString.c_str();
```
Comment (ChristianTackeGSI) [inline examples/MQ/pixelSimSplit/src/devices/FairMQTransportDevice.cxx:157]: Maybe?

```suggestion
            LOG(info) << " -> " << repString;
```

File: parmq/ParameterMQServer.cxx [hunk 5]
```diff
@@ -187,7 +187,7 @@ bool ParameterMQServer::ProcessUpdate(fair::mq::MessagePtr& update, int /*index*
         uint runId = 0;
         if (parDescr.find("RUNID") != std::string::npos) {
             parDescr.erase(0, parDescr.find("RUNID") + 5);
-            runId = atoi(parDescr.data());
+            runId = atoi(parDescr.c_str());
```
Comment (ChristianTackeGSI) [inline parmq/ParameterMQServer.cxx:190]: Could yuo take a look into [`std::stoi`](https://en.cppreference.com/w/cpp/string/basic_string/stol)?

```suggestion
            runId = std::stoi(parDescr);
```

**Review by ChristianTackeGSI — COMMENTED** [review]
Please also take a look at the commit message.

**Review by ChristianTackeGSI — APPROVED** [review]

**Pull request merged**